stdlib/_socket.pyi

```diff
@@ -0,0 +1,627 @@
+import sys
+from _typeshed import WriteableBuffer
+from collections.abc import Iterable
+from typing import Any, Optional, SupportsInt, Tuple, Union, overload
+
+if sys.version_info >= (3, 8):
```

Comment:
I was a bit surprised by this but it seems right; PyLong_AsSocket_t is an alias for PyLong_AsLong, and that does `__index__` on main but not on 3.7.

Reply:
I'd like to claim that was intentional, but it was pure lazyness on my part, since I didn't want to write a custom protocol. Of course, I could have imported from typing_extensions ...